hpct-ldap-server/src/managers/openldap.py

```diff
@@ -0,0 +1,491 @@
+            ldap_uri = subprocess.run(["cat", "/etc/hostname"], capture_output=True, text=True)
+            sssd_conf = [
+                "[sssd]",
+                "config_file_version = 2",
+                f"domains = {domain}",
+                "",
+                f"[domain/{domain}]",
+                "id_provider = ldap",
+                "auth_provider = ldap",
+                f"ldap_uri = ldap://{ldap_uri.stdout}",
+                "cache_credentials = True",
+                f"ldap_search_base = {dcs}",
+            ]
+            sssd_conf = "\n".join(sssd_conf)
+            with open("/etc/sssd/sssd_conf.template", "w") as f:
+                f.write(f"{sssd_conf}")
+
+            fd = FileData()
+            fd.load("/etc/sssd/sssd_conf.template")
+            sssd_conf = fd._dumps()
+            fd.load("/usr/local/share/ca-certificates/mycacert.crt")
+            ca_cert = fd._dumps()
+
+            return ca_cert, sssd_conf
+        else:
+            raise Exception("Domain has not been set. Run action set-config first.")
+
+    def configure(self, admin_passwd=None, domain=None, org=None) -> None:
+        """Set LDAP password and slapd configuration.
+
+        Parameters
+        ----------
+        admin_passwd : str
+            LDAP password.
+        domain : str
+            Domain name.
+        org : str
+            Organization name.
+        """
+        if domain and org and admin_passwd:
+            dcs = self._split_domain(domain)
+
+            # Reconfigure slapd with password
+            self.slapd_config(admin_passwd, domain, org)
+            self._add_base(admin_passwd, dcs)
+        else:
+            raise Exception("Domain, Organization, or Password is missing cannot complete action.")
+
+    def disable(self) -> None:
+        """Disable services."""
+        for name in self.systemd_services:
+            systemd.service_pause(name)
+
+    def enable(self) -> None:
+        """Enable services."""
+        for name in self.systemd_services:
+            systemd.service_resume(name)
+
+    def install(self) -> None:
+        """Install packages."""
+        if self.packages:
+            try:
+                apt.update()
+                for name in self.packages:
+                    apt.add_package(name)
+            except:
+                raise Exception(f"failed to install package ({name})")
+
+    def is_enabled(self) -> bool:
+        """Check enabled status of services."""
+        if self.systemd_services:
+            for name in self.systemd_services:
+                if not systemd._systemctl("is-enabled", name, quiet=True):
+                    return False
+
+        return True
+
+    def is_installed(self) -> bool:
+        """Check packages are installed."""
+        if self.packages:
+            for name in self.packages:
+                if not apt.DebianPackage.from_installed_package(name).present:
+                    return False
+
+        return True
```

Comment:
See comment for this method in the client code.

Reply:
Updated